scripts/templates-builder/src/generateHtmlTemplates.js

```diff
@@ -0,0 +1,136 @@
+const path = require("node:path");
+const ejs = require("ejs");
+const fs = require("fs-extra");
+const {
+  BASE_SOURCE_DIR,
+  loadTranslations,
+  BASE_OUTPUT_DIR,
+  getOutputFileName,
+} = require("./utils");
+const log = require("./logger");
+
+const templatesDir = path.join(BASE_SOURCE_DIR, "templates");
+
+async function generateHtmlTemplate() {
+  log.info("...generating html templates");
+
+  try {
+    // clear the output directory
+    await fs.emptyDir(path.join(BASE_OUTPUT_DIR, "templates"));
+
+    const entries = await fs.readdir(templatesDir, { withFileTypes: true });
+
+    for (const entry of entries) {
+      const templateName = entry.name;
+
+      if (!entry.isDirectory() || entry.name === "partials") {
+        continue;
+      }
+
+      const templateDir = path.join(templatesDir, templateName);
+
+      // Copy index.txt if exists
+      if (fs.pathExistsSync(path.join(templateDir, "index.txt"))) {
+        const outputFile = path.join(
+          BASE_OUTPUT_DIR,
+          "templates",
+          templateName,
+          `${templateName}.txt`
+        );
+
+        log.debug(`Writing ${outputFile}`);
+        await fs.copy(path.join(templateDir, "index.txt"), outputFile);
+        continue;
+      }
+
+      if (!fs.pathExistsSync(path.join(templateDir, "index.html"))) {
+        continue;
+      }
+
+      const i18nDir = path.join(templateDir, "i18n");
+
+      const templateContent = await fs.readFile(
+        path.join(templateDir, "index.html"),
+        "utf8"
+      );
+
+      const hasLanguages = fs.existsSync(i18nDir);
+
+      if (!hasLanguages) {
+        const outputFileName = getOutputFileName(templateName);
+        await renderAndWriteFile({
+          templateName,
+          fileName: outputFileName,
+          content: templateContent,
+        });
+        continue;
+      }
+
+      const languages = (
+        await fs.readdir(i18nDir, { withFileTypes: true })
```

Comment:
I have a doubt. Do we need to read files with extensions and after remove it? Can we read directly the files without extensions?

Reply:
Ok, `withFileTypes` is not needed. We read only the file name. I removed it